warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/TableSplitsCache.java

```diff
@@ -2,30 +2,51 @@
 
 import java.io.BufferedOutputStream;
 import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
 import java.io.FileNotFoundException;
 import java.io.IOException;
+import java.io.InputStreamReader;
 import java.io.PrintStream;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Properties;
 import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
 
-import org.apache.accumulo.core.client.AccumuloClient;
+import org.apache.accumulo.core.client.Accumulo;
 import org.apache.accumulo.core.client.AccumuloException;
 import org.apache.accumulo.core.client.AccumuloSecurityException;
 import org.apache.accumulo.core.client.TableNotFoundException;
-import org.apache.accumulo.core.client.admin.TableOperations;
+import org.apache.accumulo.core.client.security.tokens.PasswordToken;
+import org.apache.accumulo.core.clientImpl.ClientConfConverter;
```

Comment:
Note that these classes are marked as being in the private Accumulo API. We have been trying to reduce/remove references to private API where possible. I had already cong through and removed Threads references before (and in fact in newer versions of 2.1.3 the api has changed and this will likely break).

You can see the build warnings for import control with a regular maven build:

```
[WARN] warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/TableSplitsCache.java:27:1: Disallowed import - org.apache.accumulo.core.clientImpl.ClientConfConverter. [ImportControl]
[WARN] warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/TableSplitsCache.java:28:1: Disallowed import - org.apache.accumulo.core.clientImpl.ClientContext. [ImportControl]
[WARN] warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/TableSplitsCache.java:29:1: Disallowed import - org.apache.accumulo.core.clientImpl.ClientInfo. [ImportControl]
[WARN] warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/TableSplitsCache.java:31:1: Disallowed import - org.apache.accumulo.core.dataImpl.KeyExtent. [ImportControl]
[WARN] warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/TableSplitsCache.java:32:1: Disallowed import - org.apache.accumulo.core.metadata.MetadataServicer. [ImportControl]
[WARN] warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/TableSplitsCache.java:33:1: Disallowed import - org.apache.accumulo.core.singletons.SingletonReservation. [ImportControl]
[WARN] warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/TableSplitsCache.java:34:1: Disallowed import - org.apache.accumulo.core.util.threads.Threads. [ImportControl]
```

Reply:
oh, let me take a look at that

Reply:
Looks like it at least doesnt break after building with new Accumulo jars. The breaking change was in ThreadPools not Threads, but it would be good to remove what we can. If there are things here that cannot be removed or that should be public in accumulo we should start that discussion with the accumulo team and add the class to import-control-accumulo.xml

Reply:
I don't think there is a public API to do this currently.  I posted a question in ASF slack to see what I can do.  I suppose I could simply scan the metadata table but I would really prefer a better way.

Reply:
I should be able to remove the use of Threads at least.

Reply:
Actually the accumulo client is a client context, so I can remove a bunch of it.  However I cannot get rid of the metadata servicer.